File: projects/10/ruby/Kiwi.rb
# Kiwi.rb | Defines the Kiwi class which inherits attributes and methods
#   from the Bird superclass.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Quentin Barnes
# Date: May 07, 2019
####################################################

require './WalkingBird.rb'

class Kiwi < WalkingBird

    def call
        'Screech!'
    end

end

File: projects/10/java/Kiwi.java
/* Kiwi.java provides a Kiwi subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Quentin Barnes
 * Date: May 7, 2019
 ******************************************************/

public class Kiwi extends WalkingBird {

  /*
   * explicit constructor Receive: name, a String PostCond: myName == name.
   */

  public Kiwi() {
    super();
  }

  public Kiwi(String name) {
    super(name);
  }

  /*
   * A Kiwi's Call Return: a Kiwi-call ("Quack!").
   */

  public String call() {
    return "Screech!";
  }

}
